notify/notify.go

```diff
@@ -293,72 +375,48 @@ func (n *DedupingNotifier) Notify(ctx context.Context, alerts ...*types.Alert) e
 		})
 	}
 
-	if err := n.notifier.Notify(ctx, alerts...); err != nil {
-		return err
-	}
-
-	return n.notifies.Set(newNotifies...)
+	return newNotifies, nil
 }
 
-type WaitNotifier struct {
-	wait     func() time.Duration
-	notifier Notifier
+// WaitStage waits for a certain amount of time before continuing or until the
+// context is done.
+type WaitStage struct {
+	wait func() time.Duration
 }
```

Comment:
This can actually be `type WaitStage func() time.Duration`. 
But more of a taste thing than relevant :)